server/messages.go

```diff
@@ -131,7 +132,10 @@ func processMessages(conn net.Conn, c chan *packetInfo, t *Tracker) {
 
 			if b.Len() >= numReadBytes {
 				msg := make([]byte, numReadBytes)
-				b.Read(msg)
+				if _, err := b.Read(msg); (err != nil) && (err != io.EOF) {
+					fmt.Fprintf(os.Stderr, "[Error] reading from message buffer: %s\n", err)
```

Comment:
I think this is robust but it's worth checking.

Reply:
Definitely. 👍

Reply:
This was robust but wrong... At this point, the parser definitely needs more data. If it is empty with `EOF`, that is also an error.

Reverted in f74887f